File: backends/php-laravel/app/Domain/Notification/ValueObjects/NotificationCountsVo.php
<?php
/**
 * This is a program developed by BobTabo.
 *
 * Copyright (c) 2026 BobTabo. All Rights Reserved.
 */
namespace App\Domain\Notification\ValueObjects;

use App\Support\Traits\Getter;
use App\Support\ValueObjects\AbstractValueObject;

/**
 * 通知件数集計の結果 ValueObject です。
 *
 * @author Satoshi Nagashiba <satoshi.nagashiba@gmail.com>
 * @package App\Domain\Notification\ValueObjects
 *
 * @method int getUnread() 未読通知の件数を返します。
 * @method int getTotal() 全通知の件数を返します。
 * @method array getCounts() タイプ別の通知件数を返します。
 */
class NotificationCountsVo extends AbstractValueObject
{
    use Getter;

    private int $unread = 0;
    private int $total = 0;
    private array $counts = [];
}

File: backends/java-spring/src/main/java/com/authorization/domain/notification/valueobjects/NotificationCountsVo.java
/**
 * This is a program developed by BobTabo.
 *
 * Copyright (c) 2026 BobTabo. All Rights Reserved.
 */
package com.authorization.domain.notification.valueobjects;

import com.authorization.support.valueobjects.AbstractValueObject;
import java.util.Map;
import lombok.Data;
import lombok.EqualsAndHashCode;

/**
 * 通知件数ValueObjectクラスです。
 *
 * @author Satoshi Nagashiba <satoshi.nagashiba@gmail.com>
 */
@Data
@EqualsAndHashCode(callSuper = true)
public class NotificationCountsVo extends AbstractValueObject {

    private int unread;
    private int total;
    private Map<String, Integer> counts = Map.of();
}
